llvm/lib/CAS/PluginCAS.cpp

```diff
@@ -422,6 +425,15 @@ Error PluginObjectStore::pruneStorageData() {
   return Error::success();
 }
 
+Error PluginObjectStore::validate(bool CheckHash) const {
+  if (Ctx->Functions.cas_validate) {
+    char *c_err = nullptr;
+    if (Ctx->Functions.cas_validate(Ctx->c_cas, CheckHash, &c_err))
```

Comment:
if this is false (no error) it falls through to reporting the `StringError` below.

Reply:
Nice catch.